pkgs/build-support/nix-gitignore/default.nix

```diff
@@ -41,6 +41,9 @@ in rec {
         let split = match "^(!?)(.*)" l;
         in [(elemAt split 1) (head split == "!")];
 
+      # regex -> regex
+      handleHashes = replaceStrings ["\\#"] ["#"];
```

Comment:
Shouldn't this only be done for the first character of the line? (based on `man gitignore`)

Reply:
Empirically, if I add `foo\#bar` to .gitignore and create a file called `foo#bar` it is ignored.

Reply:
Can you also try. `foo\#bar\#baz`? I see that some lines in the docs say only the first one can be escaped. Sorry for the nitpick.

Reply:
If I create a file `foo#bar#baz` and add `foo\#bar\#baz` to .gitignore, the file is ignored by both git and nix-gitignore.